Fix checks2

mblue9 opened the pull request.

mblue9 committed:
Delete pbmc_small tidySCE carryover

mblue9 committed:
Remove DESeq carryover and fix test

mblue9 committed:
package should be imported in the NAMESPACE file

Made edits here to fix this:
* Checking if other packages can import this one...
    * ERROR: Packages providing 2 object(s) used in this
      package should be imported in the NAMESPACE file,
      otherwise packages importing this package may
      fail.

      package::object in function()
        dplyr::n in is_not_duplicated()
        dplyr::n in is_rectangular()

mblue9 commented:
do you know how to get past this error?
```
tt <- pasilla %>% tidy()
tt %>% filter(condition == "untreated")
Error: Problem with `mutate()` input `..1`.
x Input `..1` must be a vector, not a function.
ℹ Input `..1` is `dplyr::n`.
Run `rlang::last_error()` to see where the error occurred.

stemangiola commented:
> 
> 
> do you know how to get past this error?
> 
> ```
> tt <- pasilla %>% tidy()
> tt %>% filter(condition == "untreated")
> Error: Problem with `mutate()` input `..1`.
> x Input `..1` must be a vector, not a function.
> ℹ Input `..1` is `dplyr::n`.
> Run `rlang::last_error()` to see where the error occurred.
> ```

yes using tidySE::filter. But that's not good, I don't know why for this package dplyr overrides our functions. using tidySE should not be needed. I have to investigate

stemangiola committed:
try to fix errors CHECK

stemangiola committed:
try to fix errors

mblue9 committed:
Fix ggplot in vignette

mblue9 committed:
Remove unneeded figure

mblue9 committed:
Remove unused Suggests

mblue9 committed:
Fix BiocCheck error package should be imported in the NAMESPACE file

Made edits here to fix this:
* Checking if other packages can import this one...
    * ERROR: Packages providing 2 object(s) used in this
      package should be imported in the NAMESPACE file,
      otherwise packages importing this package may
      fail.

      package::object in function()
        dplyr::n in is_not_duplicated()
        dplyr::n in is_rectangular()

stemangiola commented on R/validation.R:
```diff
@@ -4,14 +4,14 @@ is_rectangular <- function(.data){
     .data %>%
     count(sample) %>%
-    count(`n`, name = "nn") %>%
+    count(dplyr::n, name = "nn") %>%
```
the n here is a column name, not a function. Any particular reson you are using this form? Daos it fail on your computer otherwise?

mblue9 replied:
I put it in the commit description, with just `n` it throws an error with BiocCheck. This is one way to fix the error do you know any other?

mblue9 replied:
actually I can just add `@importFrom dplyr n` instead

stemangiola replied:
> 
> 
> I put it in the commit description, with just `n` it throws an error with BiocCheck. This is one way to fix the error do you know any other?

Strange it does not on my biocheck. Do you have Bioconductor 3.12 or 3.11 installed?

mblue9 replied:
3.12. What BiocCheck version do you have? I updated to latest 1.25.5

stemangiola replied:
This is BiocCheck version 1.25.8. and I did not get error. so strange :)

mblue9 replied:
Yes strange, I will test my end with BiocCheck version 1.25.8. Yep good to use 3.12 as that's what we're supposed to develop against.

mblue9 replied:
Ok that's interesting, no error with BiocCheck 1.25.8. But 1.25.8. is the GitHub biocheck version that's not released yet & I guess they might use the Bioc devel version to test packages? which is 1.25.5 https://bioconductor.org/packages/devel/bioc/html/BiocCheck.html
should I remove the change if we think 1.25.8 will be what they use to check and if it throws the error we say it works with 1.25.8?

mblue9 committed:
Add @importFrom dplyr n

mblue9 commented:
Do you know why the rcmdcheck keeps failing here in the ubuntu check with
```
Error: C stack usage  7969220 is too close to the limit
Execution halted
Error in proc$get_built_file() : Build process failed
Calls: <Anonymous> ... build_package -> with_envvar -> force -> <Anonymous>
Execution halted
##[error]Process completed with exit code 1.
```
The rcmdcheck runs without error for me locally (just 1 note) but I've OSX so maybe something different with Linux?

mblue9 commented:
maybe it's due to updates in packages used by tidySE? https://www.mail-archive.com/bioc-devel@r-project.org/msg11420.html

stemangiola commented:
> maybe it's due to updates in packages used by tidySE? https://www.mail-archive.com/bioc-devel@r-project.org/msg11420.html

so strange

```
Error: C stack usage  7969220 is too close to the limit
Execution halted
Error in proc$get_built_file() : Build process failed
Calls: <Anonymous> ... build_package -> with_envvar -> force -> <Anonymous>
Execution halted
```

we don't use singlecellexperiment

mblue9 commented:
the link I put is just an example of what can cause an error like that. might be due to some change to summarizedexperiment or other package that tidySE uses?

stemangiola commented:
Thanks @mblue9! I will submit. I just wat to see what error it causes on the Bioc servers for submission.
